### ISSUE TITLE
Release Geopackage file

### ISSUE THREAD

**Issue by SouthEndMusic** [Deltares/Ribasim#269]
I've noticed that when the Ribasim core is run from a REPL (e.g. `Ribasim.run("path/to/model.toml")`, the geopackage file is not released after this call. This is a bit of a nuisance when you want to make small changes to the model and write it again, since you cannot do this when the Julia REPL session is still active. Ideally the geopackage file would even be released when the Julia core throws an error.

**Comment by visr** [comment]
Do you have a reproducible example? If I run `Ribasim.run("data/basic/basic.toml")` the GeoPackage is released as far as I can tell. I can run `python python\ribasim\tests\conftest.py` without having to close my Julia session.

This is expected since at the end of the initialize function the GeoPackage is closed:
https://github.com/Deltares/Ribasim/blob/main/core/src/bmi.jl#L49

That's not to say that we cannot clear up better, I've had issues with locked files as well, though not sure how to reproduce. If the model initialization now fails between opening and closing the GeoPackage, then it can stay open. To fix that I think we need to put that code in a try finally block, like below:

```julia
f = open("file")
try
    operate_on_file(f)
finally
    close(f)
end
```

**Comment by SouthEndMusic** [comment]
Your suggestion sounds good. It could indeed be that a run without errors releases the file and I only tested it for runs that gave errors in the section which operates on the file.

**Comment by evetion** [comment]
According to the [documentation](https://juliadatabases.org/SQLite.jl/stable/#SQLite.DB), DB is automatically closed?

**Comment by visr** [comment]
Yes, if it goes out of scope on the next GC. Though better to not rely on that, since you don't know exactly when that is.

**Comment by evetion** [comment]
Partially fixed by #271 

> I tried it but now I found that I could not delete a generated geopackage file because of the Python REPL from which I wrote it.
